File: Cap 10/10.6 - Passagens de Parametro/10.6.1 - Passagem de Parametro por Valor/calcfat1.php
<?php
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 10 - Topico 10.6.1
 * 
 * Programa ..: PASSAGEM DE PARAMETRO POR VALOR
 */

function fatorial($N) {
    $FAT = 1;
    for ($I = 1; $I <= $N; $I++) {
        $FAT = $FAT * $I;
    }
    echo "Fatorial = $FAT\n";
}

echo "CALCULO DE FATORIAL\n";
echo "PROCEDIMENTO COM PASSAGEM DE PARAMETRO POR VALOR\n";
echo "\n";

echo "Qual fatorial (1-20): ";
$LIMITE = intval(readline());

fatorial($LIMITE);

?>

File: Cap 10/10.6 - Passagens de Parametro/10.6.1 - Passagem de Parametro por Valor/calcfat1.kt
/*
 * Livro .....: Algoritmos
 *              Lógica para Desenvolvimento de Programação de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.6.1
 *
 * Programa ..: PASSAGEM DE PARAMETRO POR VALOR
 */

import java.util.Scanner

fun fatorial(N: Int) {
    var FAT: Long = 1
    for (I in 1..N) {
        FAT = FAT * I
    }
    println("Fatorial = $FAT")
}

fun main() {
  
    var entrada = Scanner(System.`in`)

    println("CALCULO DE FATORIAL")
    println("PROCEDIMENTO COM PASSAGEM DE PARAMETRO POR VALOR\n")

    print("Qual fatorial (1-20): ")
    var LIMITE = entrada.nextInt()

    fatorial(LIMITE)
	
}
